cli/dcos-spark/submit_builder.go

```diff
@@ -328,6 +334,21 @@ func setupTLSArgs(args *sparkArgs) {
 	}
 }
 
+func setupSaslProperties(secretPath string, args *sparkArgs) {
+	args.properties["spark.mesos.containerizer"] = "mesos"
+	args.properties["spark.authenticate"] = "true"
+	args.properties["spark.authenticate.enableSaslEncryption"] = "true"
+	args.properties["spark.authenticate.secret"] = "spark_shared_secret"
+	args.properties["spark.executorEnv._SPARK_AUTH_SECRET"] = "spark_shared_secret"
+	taskTypes := []string{"driver", "executor"}
+	for _, taskType := range taskTypes {
+		secretRefProp := fmt.Sprintf(SECRET_REFERENCE_TEMPLATE, taskType)
+		secretFileProp := fmt.Sprintf(SECRET_FILENAME_TEMPLATE, taskType)
+		appendToProperty(secretRefProp, secretPath, args)
+		appendToProperty(secretFileProp, secretPath, args)
```

Comment:
It won't create a sub directory for the secret. The directory structure for the `reference` is a construct for permissions, and is related to `DCOS_SPACE`. So if the secret `reference` was `userA/keytab` and the `DCOS_SPACE` for the framework was `/userB` you would not have access to the secret (it would need to be `/userA`). 

However you bring up that there is a bug here, and in the handling of secret references in general. I'm going to clean it up in another PR (https://jira.mesosphere.com/browse/SPARK-573 and https://jira.mesosphere.com/browse/SPARK-620).